crates/ide-assists/src/utils.rs

```diff
@@ -94,16 +94,44 @@ pub fn filter_assoc_items(
     sema: &Semantics<'_, RootDatabase>,
     items: &[hir::AssocItem],
     default_methods: DefaultMethods,
+    ignore_hidden: bool,
 ) -> Vec<InFile<ast::AssocItem>> {
+    // FIXME : How to use the closure below  this so I can write sth like
+    // sema.source(hide(it)?)?...
+
+    // let hide = |it: impl HasAttrs| {
+    //     if default_methods == DefaultMethods::IgnoreHidden && it.attrs(sema.db).has_doc_hidden() {
+    //         None;
+    //     }
+    //     Some(it)
+    // };
+
     return items
         .iter()
         // Note: This throws away items with no source.
         .copied()
         .filter_map(|assoc_item| {
             let item = match assoc_item {
-                hir::AssocItem::Function(it) => sema.source(it)?.map(ast::AssocItem::Fn),
-                hir::AssocItem::TypeAlias(it) => sema.source(it)?.map(ast::AssocItem::TypeAlias),
-                hir::AssocItem::Const(it) => sema.source(it)?.map(ast::AssocItem::Const),
+                hir::AssocItem::Function(it) => {
+                    if ignore_hidden && it.attrs(sema.db).has_doc_hidden() {
+                        return None;
+                    }
+                    sema.source(it)?.map(ast::AssocItem::Fn)
+                }
```

Comment:
we can use match guards here
```suggestion
                hir::AssocItem::Function(it) if !it.attrs(sema.db).has_doc_hidden() => {
                    sema.source(it)?.map(ast::AssocItem::Fn)
                }
```
And then put a wild card arm at the end that `_ => return None,`

Reply:
I think these can be simplified further (subjective yes :wink:). Since `hir::AssocItem` implements `HasAttrs`, we can keep current `filter_map()` and use `filter()` before it, like so:

```rust
    return items
        .iter()
        // Note: This throws away items with no source.
        .copied()
        .filter(|assoc_item| !(ignore_hidden && assoc_item.attrs(semf.db).has_doc_hidden()))
        .filter_map(|assoc_item| {
            // current impl
        });
```

Also, can you move that `Note` comment just before or inside `filter_map()`? I've misplaced it in my other PR :sweat_smile: